pkg/client/create_builder.go

```diff
@@ -212,20 +219,75 @@ func (c *Client) fetchLifecycle(ctx context.Context, config pubbldr.LifecycleCon
 	var uri string
 	var err error
 	switch {
+	case buildpack.HasDockerLocator(config.URI):
```

Comment:
Maybe we also need to validate the given URI is a valid image reference we do this when checking buildpacks

```go
if _, err := name.ParseReference(config.URI); err == nil {
     // the URI is a valid image reference.
}
```